engine/src/main/resources/swagger/cromwell.yaml

```diff
@@ -716,6 +756,208 @@ definitions:
+          - "We might also provide warnings to a 'valid' workflow here"
+          - "Otherwise, 'errors' will be empty or unspecified"
+        description: The set of validation failure messages
+      name:
+        type: string
+        description: The name of the workflow in the description file
+        example: my_wf
+      inputs:
+        description: A list of inputs for this tool
+        type: array
+        items:
+          $ref: '#/definitions/ToolParameter'
+        example:
+          - name: my_wf.string_input
+            valueType:
+              typeName: String
+            optional: false
+          - name: my_wf.array_input
+            valueType:
+              typeName: Array
+              arrayType:
+                typeName: String
+            optional: false
+          - name: my_wf.optional_input
+            valueType:
+              typeName: Optional
+              optionalType:
+                typeName: String
+            optional: true
+            default: "hello"
+          - name: my_wf.map_input
+            valueType:
+              typeName: Map
+              mapType:
+                keyType:
+                  typeName: String
+                valueType:
+                  typeName: Int
+            optional: false
+          - name: my_wf.object_input
+            valueType:
+              typeName: Object
+              objectFieldTypes:
+                - fieldName: "int_field"
+                  fieldType:
+                    typeName: Int
+                - fieldName: "int_array_field"
+                  fieldType:
+                    typeName: Array
+                    arrayType:
+                      typeName: Int
+            optional: false
+          - name: my_wf.int_string_pair_input
+            valueType:
+              typeName: Tuple
+              tupleTypes:
+                - typeName: Int
+                - typeName: String
+            optional: false
+      images:
+        type: array
+        items:
+          type: string
+          example:
+          - quay.io/seqware/seqware_full/1.1
+          - ubuntu:latest
+          - EXPRESSION_BASED_IMAGE
+        description: >-
+          All known docker images (and versions/hashes) used by this tool.
+          If the tool will have to evaluate one or more docker image strings at runtime, the placeholder
+          "EXPRESSION_BASED_IMAGE" will be used.
+      submitted_descriptor_type:
+        $ref: '#/definitions/DescriptorTypeAndVersion'
+      imported_descriptor_types:
+        type: array
+        items:
+          $ref: '#/definitions/DescriptorTypeAndVersion'
+        description: The set of all types imported by the submitted workflow.
+      meta:
+        type: object
+        description: An open set of key/value pairs echoing any workflow-level metadata found in the description file.
+        example:
+          author: "Batman"
+          copyright: "BSD 3-clause"
```

Comment:
where are the workflow `outputs`?

Reply:
Good point! Is that something that you'll need? If so, we'll have to add something for `outputs`, presumably equivalent to `inputs`?

Reply:
Yes, Terra shows the outputs of a workflow on its "method configuration" screen

See for example https://github.com/DataBiosphere/saturn-ui/issues/423

Reply:
agreed - yes, we need the outputs. I imagine it would be equivalent to the inputs section.

Reply:
In my mind, the only thing that may or may not apply to "outputs" is the field `images`. My mental abstraction is such that I think of both docker & inputs to be the left side of the equation and outputs on the right...
docker + inputs => outputs